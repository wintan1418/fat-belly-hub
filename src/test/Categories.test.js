import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Categories from '../component/Category';

describe('Creates a component', () => {
  const middlewares = [thunk];
  const initialState = { categories: [{ categ: 'miscalleneous' }] };
  const mockStore = configureStore(middlewares);
  let store;
  it('returns correctly', () => {
    store = mockStore(initialState);
    const tree = renderer.create(
      <Provider store={store}>
        <Categories />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders a component', () => {
    store = mockStore(initialState);
    render(<Provider store={store}><Categories /></Provider>);
    const element = screen.getByText('Explore!');
    expect(element).toBeInTheDocument();
  });
});
