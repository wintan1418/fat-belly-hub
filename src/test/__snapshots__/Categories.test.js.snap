// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a component returns correctly 1`] = `
<div
  className="categ"
>
  <h3>
    Explore! 
  </h3>
  <select
    className="cat"
    onChange={[Function]}
    placeholder="Select your category"
  />
</div>
`;
